src/shared/lib/OncoKbUtils.ts

```diff
@@ -184,8 +185,52 @@ export function normalizeLevel(level:string):string|null
     }
 }
 
-export function oncogenicImageClassNames(oncogenic:string,
-                                         isVUS:boolean,
+export function oncogenicXPosition(highestSensitiveLevel: string | null) {
+    const map: { [id: string]: number } = {
+        '1': 1,
+        '2A': 2,
+        '2B': 3,
+        '3A': 4,
+        '3B': 5,
+        '4': 6
+    };
+
+    let levelIndex = highestSensitiveLevel === null ? 0 : (map[highestSensitiveLevel] || 0);
+    return -(8 + levelIndex * 30);
+}
+
+export function oncogenicYPosition(oncogenicity: string, isVUS: boolean, resistanceLevel: string | null) {
+    const oncogenicityMap: { [id: string]: number } = {
+        'Oncogenic': 0,
+        'Predicted Oncogenic': 0,
+        'Likely Oncogenic': 0,
+        'Neutral': 1,
+        'Likely Neutral': 1,
+        'Unknown': 2,
+        'Inconclusive': 2,
+        '': 4
+    };
+    const resistanceLevelMap: { [id: string]: number } = {
+        '': 0,
+        'R1': 1,
+        'R2': 2,
+        'R3': 3
+    };
+
+    let oncogenicityIndex = oncogenicityMap[oncogenicity];
+    if (oncogenicityIndex === undefined) {
+        oncogenicityIndex = 4;
+    }
+    if (oncogenicityIndex > 1 && isVUS) {
+        oncogenicityIndex = 3;
+    }
+
+    let resistanceLevelIndex = resistanceLevel === null ? 0 : (resistanceLevelMap[resistanceLevel] || 0);
```

Comment:
seems like maybe you don't need that '' : 0 in the resistanceLevelMap because it's a default.   so, i think all you need is just `resistanceLevelMap[resistanceLevel] || 0`

Reply:
Still need to do a null check though